src/tree/EnvironmentTreeItem.ts

```diff
@@ -91,7 +91,18 @@ export class EnvironmentTreeItem extends AzureParentTreeItem implements IAzureRe
         return treeUtils.getIconPath('Azure-Static-Apps-Environment');
     }
 
-    public async loadMoreChildrenImpl(_clearCache: boolean, _context: IActionContext): Promise<AzExtParentTreeItem[]> {
+    public async loadMoreChildrenImpl(_clearCache: boolean, _context: IActionContext): Promise<AzExtTreeItem[]> {
+        const client: WebSiteManagementClient = createAzureClient(this.root, WebSiteManagementClient);
+        const functions: WebSiteManagementModels.StaticSiteFunctionOverviewCollection = await client.staticSites.listStaticSiteBuildFunctions(this.parent.resourceGroup, this.parent.name, this.buildId);
```

Comment:
Hmm there's not a simpler API we can call to detect if they have functions set up? For example, technically I could have a function app, but just no functions. Would be nice if there's just like a property on the site config or something

Reply:
As far as I know, SWA only creates a function app if there are functions in the designated API folder.  If you don't have functions, it'll give you the error that a function must be first deployed.  Even if you previously had a function and then remove the api folder, it'll go back to throwing that error.

I checked the properties on the build itself, but don't see anything that we could use for that.

Reply:
I could use the list function app settings API call if you think that'd be faster.  It'll throw an error if it can't detect a function app, so in case the scenario you described is possible, it should be a more reliable indicator of whether or not there's a function app.

Reply:
🤷‍♂️